internal/pkg/plugininstaller/github/option.go

```diff
@@ -0,0 +1,78 @@
+package github
+
+import (
+	"fmt"
+
+	"github.com/devstream-io/devstream/pkg/util/template"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/pkg/util/github"
+)
+
+const (
+	RegistryDockerhub   RegistryType = "dockerhub"
+	RegistryHarbor      RegistryType = "harbor"
+	CommitMessage                    = "GitHub Actions workflow, created by DevStream"
+	PRBuilderFileName                = "pr-builder.yml"
+	MainBuilderFileName              = "main-builder.yml"
+)
+
+// GithubActionLanguage is the language of repo
+type GithubActionLanguage struct {
+	Name    string `validate:"required"`
+	Version string
+}
+
+// GithubActionOptions is the struct for configurations of the  plugin.
+type GithubActionOptions struct {
+	Owner    string                `validate:"required_without=Org"`
+	Org      string                `validate:"required_without=Owner"`
+	Repo     string                `validate:"required"`
+	Branch   string                `validate:"required"`
+	Language *GithubActionLanguage `validate:"required"`
+
+	// optional
+	Workflows []*github.Workflow
+	Build     *Build
+	Test      *Test
+	Docker    *Docker
+}
+
+func NewGithubActionOptions(options plugininstaller.RawOptions) (*GithubActionOptions, error) {
+	var opts GithubActionOptions
+	if err := mapstructure.Decode(options, &opts); err != nil {
+		return nil, err
+	}
+	return &opts, nil
+}
+
+func (opts *GithubActionOptions) GetLanguage() string {
+	return fmt.Sprintf("%s-%s", opts.Language.Name, opts.Language.Version)
+}
+
+func (opts *GithubActionOptions) GetGithubClient() (*github.Client, error) {
+	ghOptions := &github.Option{
+		Owner:    opts.Owner,
+		Org:      opts.Org,
+		Repo:     opts.Repo,
+		NeedAuth: true,
+	}
+	return github.NewClient(ghOptions)
+}
+
+func (opts *GithubActionOptions) RenderWorkFlow(content string) (string, error) {
+	if opts.Build == nil {
+		opts.Build = &Build{false, ""}
```

Comment:
If the default value of `Build` is zero value for each field, you can reduce this code by changing `Build *Build` to `Build Build`.